Core/src/org/sleuthkit/autopsy/casemodule/MultiUserCaseManager.java

```diff
@@ -0,0 +1,132 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.sleuthkit.autopsy.casemodule;
+
+import java.io.File;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+import org.sleuthkit.autopsy.coordinationservice.CoordinationService;
+import org.sleuthkit.autopsy.coordinationservice.CoordinationService.CoordinationServiceException;
+
+/**
+ * Handles locating and opening multi-user cases.
+ */
+final class MultiUserCaseManager {
+
+    private static MultiUserCaseManager instance;
+    
+    private CoordinationService coordinationService;
+
+    /**
+     * Gets the multi-user case manager.
+     *
+     * @return The multi-user case manager singleton.
+     *
+     * @throws MultiUserCaseManagerException
+     */
+    synchronized static MultiUserCaseManager getInstance() throws MultiUserCaseManager.MultiUserCaseManagerException {
+        if (null == instance) {
+            instance = new MultiUserCaseManager();
+        }
+        return instance;
+    }
+
+    /**
+     * Constructs an object that handles locating and opening multi-user cases.
+     *
+     * @throws MultiUserCaseManagerException
+     */
+    private MultiUserCaseManager() throws MultiUserCaseManagerException {
+        try {
+            coordinationService = CoordinationService.getInstance();
+        } catch (CoordinationServiceException ex) {
+            throw new MultiUserCaseManager.MultiUserCaseManagerException("Failed to get the coordination service.", ex);
+        }
+    }
+
+    /**
+     * Gets a list of the cases in the top level case folder
+     *
+     * @return List of cases.
+     *
+     * @throws CoordinationServiceException
+     */
+    List<MultiUserCase> getCases() throws CoordinationServiceException {
+        List<MultiUserCase> cases = new ArrayList<>();
+        List<String> nodeList = coordinationService.getNodeList(CoordinationService.CategoryNode.CASES);
+        for (String node : nodeList) {
+                Path casePath = Paths.get(node);
+                File caseFolder = casePath.toFile();
+                if(caseFolder.exists()) {
+                        File[] autFiles = caseFolder.listFiles((dir, name) -> name.toLowerCase().endsWith(".aut"));
+                        if(autFiles != null && autFiles.length > 0) {
+                                cases.add(new MultiUserCase(casePath));
+                        }
+                }
+        }
+        return cases;
+    }
+
+    /**
+     * Opens a multi-user case.
+     *
+     * @param caseMetadataFilePath Path to the case metadata file.
+     *
+     * @throws CaseActionException
+     */
+    synchronized void openCase(Path caseMetadataFilePath) throws CaseActionException {
+        /*
+         * Open the case.
+         */
+        Case.openAsCurrentCase(caseMetadataFilePath.toString());
```

Comment:
When a case is opened via this path, it does not appear as a **Recent Cases** menu item.